File: BACK-END/KuentecoApp/src/main/java/org/kuenteco/backend/dto/logic/budget/BudgetEnrollmentDTO.java
package org.kuenteco.backend.dto.logic.budget;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BudgetEnrollmentDTO {
    @NotBlank private String userEmail;
    @NotBlank private String profileEmail;
    @NotBlank private String budgetName;
}

File: FRONT-END/lib/dto/budget/budget_enrollment_dto.dart
class BudgetEnrollmentDTO {
  final String userEmail;
  final String profileEmail;
  final String budgetName;

  BudgetEnrollmentDTO({
    required this.userEmail,
    required this.profileEmail,
    required this.budgetName,
  });

  factory BudgetEnrollmentDTO.fromJson(Map<String, dynamic> json) {
    return BudgetEnrollmentDTO(
      userEmail: json['userEmail'],
      profileEmail: json['profileEmail'],
      budgetName: json['budgetName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userEmail': userEmail,
      'profileEmail': profileEmail,
      'budgetName': budgetName,
    };
  }
}
